code/modules/power/treadmill.dm

```diff
@@ -51,19 +51,28 @@
 /obj/machinery/power/treadmill/proc/powerwalk(atom/movable/AM as mob)
 	if(!ismob(AM)) return //Can't walk on a treadmill if you aren't animated
 	if(get_turf(AM) != loc) return //Can't bump from the outside
-	var/mob/runner = AM
-	if(!istype(runner,/mob/living/simple_animal)&&runner.bodytemperature <= 360)
-		runner.bodytemperature += 2 //Same heating pattern as being fat
-	if(runner.nutrition && runner.stat != DEAD)
-		runner.nutrition -= HUNGER_FACTOR*2 //Running on a treadmill makes you hungry fast
-	flick("treadmill-running", src)
-	playsound(get_turf(src), 'sound/machines/click.ogg', 50, 1)
-	var/calc = DEFAULT_BUMP_ENERGY * power_efficiency * runner.treadmill_speed
-	if(runner.reagents) //Sanity
-		for(var/datum/reagent/R in runner.reagents.reagent_list)
-			calc *= R.sport
-	if(M_HULK in runner.mutations) calc *= 5
-	count_power += calc
+	var/mob/living/runner = AM
+	var/cached_temp = runner.bodytemperature
+	if(runner.burn_calories(HUNGER_FACTOR*2))
+		flick("treadmill-running", src)
+		playsound(get_turf(src), 'sound/machines/click.ogg', 50, 1)
+		var/calc = DEFAULT_BUMP_ENERGY * power_efficiency * runner.treadmill_speed
+		if(runner.reagents) //Sanity
+			for(var/datum/reagent/R in runner.reagents.reagent_list)
+				calc *= R.sport
+		if(M_HULK in runner.mutations) calc *= 5
+		count_power += calc
+		if(runner.bodytemperature > 99)
+			if(prob(5))
+				runner.say("*collapse")
+			if(prob(10))
+				to_chat(runner,"<span class='warning'>You really should take a rest!</span>")
+			if(prob(5))
+				to_chat(runner,"<span class='warning'>Your legs really hurt!</span>")
+				runner.apply_damage(2*damage, BRUTE, "l_leg")
+				runner.apply_damage(2*damage, BRUTE, "r_leg")
+			runner.bodytemperature = max(99,cached_temp)
```

Comment:
These temperatures are fucking awful but more realistic ones would probably ruin balance to a ridiculous degree